File: arrays/LongestCommonPrefix.java
package arrays;

public class LongestCommonPrefix {
    public String longestCommonPrefix(String[] strs) {
        
        int smallestIndex = 0;
        int smallStr = strs[0].length();
        for(int i=0;i<strs.length;i++){
            if(strs[i].equals("")) return "";
            if(strs[i].length() < smallStr){ 
                smallStr = strs[i].length(); 
                smallestIndex = i;
            }
        }

        String prefix = strs[smallestIndex];
        smallestIndex = prefix.length();
        int i=0;
        while(true){
            if(i >= strs.length) break;
            String temp = strs[i].substring(0, smallestIndex);
            if(prefix.equals(temp)) i++;
            else{ 
                if(smallestIndex <= 0) break;
                prefix = prefix.substring(0, --smallestIndex);
            }
        }
        return prefix;
    }
}

File: arrays/LongestCommonPrefix.py
from typing import List


class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        shortest_index = 0
        shortest_length = len(strs[0])
        for i in range(1,len(strs)):
            if len(strs[i]) < shortest_length:
                shortest_length = len(strs[i])
                shortest_index = i

        prefix = strs[shortest_index]
        smallest_index = len(prefix)
        i = 0
        while True:
            if i >= len(strs): break

            temp = strs[i][0:smallest_index]
            if temp == prefix: i += 1
            else:
                if smallest_index <= 0: break
                prefix = prefix[0:smallest_index-1]
                smallest_index-=1
            
        return prefix